src/System.IO.Pipelines.Networking.Windows.RIO/Internal/RioThread.cs

```diff
@@ -116,6 +116,23 @@ public IntPtr GetBufferId(IntPtr address, out long startAddress)
             return id;
         }
 
+        public unsafe RioBufferSegment GetSegmentFromMemory(Buffer<byte> memory)
+        {
+            var pin = memory.Pin();
```

Comment:
*But* since we create the pool, it's ok. Just add the comment and I'll merge it.